### PR TITLE
FINERACT 1932: Add custom Configuration class for organisation/provisioning module

### PR DESCRIPTION
**Pull request by abhinav7sinha**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by vidakovic** [comment]
@abhinav7sinha please squash the commits

**Comment by abhinav7sinha** [comment]
> @abhinav7sinha please squash the commits

The other commit wasn't supposed to be a part of this PR. I've removed it now

File: fineract-provider/src/main/java/org/apache/fineract/organisation/provisioning/service/ProvisioningCategoryReadPlatformServiceImpl.java [hunk 1]
```diff
@@ -23,18 +23,14 @@
...
-import org.springframework.stereotype.Service;
 
-@Service
 public class ProvisioningCategoryReadPlatformServiceImpl implements ProvisioningCategoryReadPlatformService {
 
     private final JdbcTemplate jdbcTemplate;
     private final ProvisioningCategoryRowMapper provisionCategoryRowMapper;
 
-    @Autowired
     public ProvisioningCategoryReadPlatformServiceImpl(final JdbcTemplate jdbcTemplate) {
```
Comment (vidakovic) [inline fineract-provider/src/main/java/org/apache/fineract/organisation/provisioning/service/ProvisioningCategoryReadPlatformServiceImpl.java]: Again, constructor can be avoided. Please use the required args annotation and instantiate the provision category row mapper in one line:
```
private final ProvisioningCategoryRowMapper provisionCategoryRowMapper = new ProvisioningCategoryRowMapper();
```

Comment (abhinav7sinha) [reply fineract-provider/src/main/java/org/apache/fineract/organisation/provisioning/service/ProvisioningCategoryReadPlatformServiceImpl.java]: done!

**Review by vidakovic — APPROVED** [review]
LGTM

**Pull request merged**